### PR TITLE
Improve usage instructions

### PR DESCRIPTION
**Pull request by OliverZim**
- minor changes in dockerfile to make the template usable
- rewritten template for better understandability
- changed devcontainer.json so the template can be used immediatly

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -75,37 +86,49 @@ For fully reproducible environments and running on HPC clusters, we provide pre-
...
+We have disabled Weights & Biases syncing with the `--offline` flag. If you want to log your results, enable W&B as described [here](#weights--biases) and omit the `--offline` flag. We also set --gpus=-1 to use all GPU's available.
 
 ### Using the Docker environment for training
 To run the training code inside the docker environment, use a `docker run` command like this:
 ```bash
-docker run --rm -it --ipc=host --gpus='"device=0,1"' -v "($pwd)":/workspace -w /workspace -v /path/to/data:/data/in/container python train.py --gpus -1 ...
+docker run -it --user $(id -u):$(id -g) --gpus='device=0' --ipc=host -v "($pwd)":/workspace -w /workspace <IMAGENAME> bash
 ```
-The `--gpus='"device=0,1"'` flag (change this to use the GPUs you actually want) selects the GPUs with indices `0` and `1` for the container and `train.py --gpus -1` makes the training script use all available GPUs (which are only the ones selected with the docker flag). 
+The `--gpus='device=0'` flag (change this to use the GPUs you actually want) selects the GPU with indice `0` for the container. Inside the container you can now execute your training-script as before.
```
Comment (konstantinjdobler) [inline README.md:129]:
```suggestion
The `--gpus='device=0'` flag (change this to use the GPUs you actually want) selects the GPU with index `0` for the container. Inside the container you can now execute your training-script as before.
```

File: scripts/console.sh [hunk 2]
```diff
@@ -12,11 +12,14 @@ if [ "$current_directory" == "scripts" ]; then
...
     --ipc host \
     --env WANDB_API_KEY \
-    -v "source=/CHANGE_ME/.cache,target=/mamba/.cache,type=bind" \
+    -v "CHANGE_ME/cache:/home/mamba/.cache" \
     -v "$(pwd)":/workspace \
     -w /workspace \
     DOCKER_TAG_CHANGE_ME \
     bash
+
+# the mounted cache folder has to exist somewhere, befor this script can be run
```
Comment (konstantinjdobler) [inline scripts/console.sh:25]:
```suggestion
# the mounted cache folder has to exist somewhere, before this script can be run
```

**Comment by Maxscha** [comment]
Can you remove the changes related to caching and create a separate PR for it? There are 1/2 other approaches which makes sense to test before we  can solve this nicely.

**Comment by Maxscha** [comment]
If you fix the merge conflicts, then it looks good to me or @konstantinjdobler

File: README.md [hunk 3]
```diff
@@ -75,37 +86,49 @@ For fully reproducible environments and running on HPC clusters, we provide pre-
...
 
 ## Development
-We provide an example setup for a remote development environment on a GPU server using [VS Code](https://code.visualstudio.com/), [Remote - SSH](https://code.visualstudio.com/docs/remote/ssh), and [Dev Containers](https://code.visualstudio.com/docs/remote/devcontainers/containers). This allows you to use the same environment for both development and production. For more details, see [here](https://code.visualstudio.com/docs/remote/advancedcontainers/develop-remote-host).
+Development for ML can be quite resource intensive. If possible, you can make use of a more powerful host machine to which you connect to with your local PC and start your development on. Normally, you would set up the correct environment on the host machine as explained above but this workflow is simplified a lot by using `VS Code Dev Containers`. They allow you to develop inside a docker container with all necessary dependencies pre-installed.  The template already contains a `.devcontainer` directory, where all the settings for it are stored, so you can start right away.
```
Comment (konstantinjdobler) [inline README.md:92]:
```suggestion
Development for ML can be quite resource intensive. If possible, you can start your development on a more powerful host machine to which you connect to from your local PC. Normally, you would set up the correct environment on the host machine as explained above but this workflow is simplified a lot by using `VS Code Dev Containers`. They allow you to develop inside a docker container with all necessary dependencies pre-installed.  The template already contains a `.devcontainer` directory, where all the settings for it are stored, so you can start right away.
```

File: README.md [hunk 4]
```diff
@@ -114,7 +137,7 @@ Weights & Biases needs access to your `WANDB_API_KEY` to be able to log results.
 For security reasons, `docker` might be disabled on your HPC cluster. You might be able to use the SLURM plugin `pyxis` instead like this:
 
 ```bash
-srun ... --container-image konstantinjdobler/nlp-research-template:torch2.0.0-cuda11.8 --container-name torch-cuda python train.py ...
+srun ... --container-image konstantinjdobler/nlp-research-template:latest --container-name torch-cuda python train.py ...
```
Comment (konstantinjdobler) [inline README.md:140]:
```suggestion
srun ... --container-image konstantinjdobler/nlp-research-template:latest python train.py ...
```

**Comment by konstantinjdobler** [comment]
I did some small re-wordings, you only need to fix the merge conflicts 😄 Some of your improvements were already on the main branch, so you can pick whichever one you like better.

**Comment by OliverZim** [comment]
I resolved the merge conflicts in the README now. The branch should be able to me merged.

**Comment by konstantinjdobler** [comment]
LGTM

**Pull request merged**